File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, arr: List[int], k: int) -> int:
        def atmost(k):
            left = 0
            ans = 0 
            count = 0
            for i in range(len(arr)):
                if arr[i] % 2 == 1:
                    count += 1
                while left < len(arr) and count > k:
                    if arr[left] % 2 == 1:
                        count -= 1
                    left += 1
                ans += i - left + 1
            return ans
        return atmost(k) - atmost(k-1)


            

File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.cpp
class Solution {
public:
    int solve(vector<int>& nums, int k){
        int n=nums.size();
        int count=0;
        int l=0;
        int ans=0;
        for(int r=0;r<n;r++){
            if(nums[r]%2)
              count++;
              while(count>=k){
                    ans+=n-r;
                    if(nums[l]%2) count--;
                    l++;
              }
        }
        return ans;
    }
    int numberOfSubarrays(vector<int>& nums, int k) {
         return solve(nums,k)-solve(nums,k+1);
    }
};